apis/v1alpha1/gateway_types.go

```diff
@@ -155,8 +152,6 @@ type Listener struct {
 	//   applied at both the TLS and HTTP protocol layers.
 	//
 	// Support: Core
-	//
-	// +required
 	Protocol ProtocolType `json:"protocol,omitempty"`
```

Comment:
Same as https://github.com/kubernetes-sigs/service-apis/pull/336/files#r492290720

Reply:
Since ProtocolType is a free-string to allow for arbitrary protocols. let us add a min and max length validation here.

Reply:
Updated, thanks!